common/djangoapps/student/models.py

```diff
@@ -916,13 +890,7 @@ def users_enrolled_in(self, course_id):
             courseenrollment__course_id=course_id,
             courseenrollment__is_active=True,
         )
-        anonymous_user_ids = [
-            user.id
-            for user in course_users
-            if not UserProfile.has_registered(user)
-        ]
-        enrolled_course_users = course_users.exclude(id__in=anonymous_user_ids)
-        return enrolled_course_users
+        return course_users
```

Comment:
We probably still want to filter out those users here;
we'll just want to check user.is_lazy_user` instead.